### PR TITLE
[charts/sn-platform] Add unified JVM options

### PR DESCRIPTION
**Pull request by fantapsody**
Fixes https://github.com/streamnative/charts/issues/573

### Motivation

Make GC/JVM configurations more structured.

### Modifications

Add `[zookeeper/bookkeeper/broker/proxy].jvm.[memoryOptions/gcOptions/extraOptions]` and keep the chart backward compatible.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below.

Need to update docs? 

- [x] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [ ] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Review by MarvinCai — APPROVED** [review]

File: charts/sn-platform/templates/_helpers.tpl [hunk 1]
```diff
@@ -105,4 +105,35 @@ Istio gateway selector
...
+{{/*
+JVM Options
+*/}}
+{{- define "pulsar.jvm.options" -}}
+jvmOptions:
+  memoryOptions:
+  {{- if .configData.PULSAR_MEM }}
+  - {{ .configData.PULSAR_MEM | quote }}
+  {{- else }}
+  {{- with .jvm.memoryOptions }}
```
Comment (FushuWang) [inline charts/sn-platform/templates/_helpers.tpl:119]:
```suggestion
  {{- with .jvm.memoryOptions -}}
```

File: charts/sn-platform/templates/_helpers.tpl [hunk 2]
```diff
@@ -105,4 +105,35 @@ Istio gateway selector
...
+  {{ toYaml . | nindent 2 }}
+  {{- end }}
+  {{- end }}
+  gcOptions:
+  {{- if .configData.PULSAR_GC }}
+  - {{ .configData.PULSAR_GC | quote }}
+  {{- else }}
+  {{- with .jvm.gcOptions }}
```
Comment (FushuWang) [inline charts/sn-platform/templates/_helpers.tpl:127]:
```suggestion
  {{- with .jvm.gcOptions -}}
```

File: charts/sn-platform/templates/_helpers.tpl (hunk 1 above)
Comment (FushuWang) [reply charts/sn-platform/templates/_helpers.tpl:119]: To delete the blank line after it.
```yaml
    jvmOptions:
      memoryOptions:

      - |
        -Xms64m -Xmx128m
      gcOptions:

      - |
        -XX:+UseG1GC
```

**Review by sijie — APPROVED** [review]

File: charts/sn-platform/templates/_helpers.tpl (hunk 1 above)
Comment (fantapsody) [reply charts/sn-platform/templates/_helpers.tpl:119]: Thanks for pointing this out, I changed `{{ toYaml` to `{{- toYaml` to align with other statements.

**Review by maxsxu — APPROVED** [review]
LGTM

**Review by FushuWang — APPROVED** [review]

**Review by ericsyh — APPROVED** [review]
LGTM

**Pull request merged**